core/src/main/java/org/apache/accumulo/core/iterators/IteratorAdapter.java

```diff
@@ -46,7 +46,7 @@ public boolean hasNext() {
       inner.next();
       return result;
     } catch (IOException ex) {
-      throw new NoSuchElementException();
+      throw new UncheckedIOException(ex);
```

Comment:
The NSEE makes a little more sense here, since it matches the Iterator interface's behavior when calling next without any more elements.